### DISCUSSION TITLE
dotLottie support

### DISCUSSION THREAD

**Discussion by lennartstolz** [airbnb/lottie-ios#1780] [Ideas]
Hey all! 

I personally think built-in `.lottie` support for `airbnb/lottie-ios` would be beneficial and could make working with Lottie animations even easier. 

## Motivation

`.lottie` are a great addition to working with Lottie and I'd love to see the `lottie-ios` project being able to open them.
I'm citing the key benefits mentioned at [dotlottie.io](https://dotlottie.io/intro/#background):

### Establishing an identity

> [...] changing the file extension from .json to .lottie avoids confusion and brings a lot of ease for designers and developers dealing with Lottie animations regularly.

### Archiving

> Storing, distributing and consuming a Lottie with its dependent assets is challenging for designers and developers, in terms of storing them on file systems and integrating into apps and websites.

### Compression

> [...] The use of archiving with compression allows all the assets required by an animation to be grouped together and the total file size to be dramatically reduced.

---

## Additional Information

- The `airbnb/lottie-android` library already does support `.lottie` files.

- There is no high-level built-in Swift/iOS/macOS framework to unpack the _dotLottie_ ZIP archive. 
   - We could make use of some of the great community ZIP frameworks: 
      - [ZIP by marmelroy](https://github.com/marmelroy/Zip) - based on minizip
      - [ZIPFoundation by weichsel](https://github.com/weichsel/ZIPFoundation) built on top of Apples `Compression` framework
    -  As the animation player would only need a fraction of the functionality those frameworks are providing and `lottie-ios` currently have no external dependencies, it could also be an option to implement the unzipping of the archives with `Compression` (which is available on macOS, iOS, tvOS, watchOS)

🙇 Looking forward to your replies!

**Comment by calda** [comment]
It would be great for this repo to support `.lottie` files by default. Definitely interested in having this functionality available.

One of the first hurdles, as you mentioned, is finding a way to unzip these files. I like the look of the [ZIPFoundation](https://github.com/weichsel/ZIPFoundation) project in particular, since it is built on top of Apple's `Compression` framework. 

Lottie currently doesn't include any third-party dependencies though, and I'm not sure how feasible it is to add one. I expect this will be easy for Swift Package Manager. I'm not sure how this works for Cocoapods though, and I think it may be difficult or impossible to do this for Carthage. I think a good first step here would be attempting to integrate ZIPFoundation into the repo and make sure that SPM, Cocoapods, and Carthage all still work correctly.

**Comment by rasberik** [comment]
Thank you for adding built in`.lottie` support.
I'd like to try to use it as soon as it will be released in next version as it helps to drastically reduce the bundle resources, but currently there is [an issue](https://github.com/airbnb/lottie-ios/pull/1802) that might result on missing animation for locally loaded `.lottie` view.

Any ideas on best approaches on how to handle it, desirably with least (or none) code change on view lifecycle after switching from `.json` to `.lottie`?

The PR above would be somewhat acceptable for my use case, at least as temporary workaround, but would nice to hear opinions on this.
